shopelectro/tests/tests_js_analytics.py

```diff
@@ -87,29 +87,34 @@ class YandexEcommerce(Ecommerce):
 
     fixtures = ['dump.json']
 
-    # @todo #799:120m Test Yandex ecommerce goals.
+    # @todo #808:120m Test Yandex ecommerce goals.
     #  Here are goals left to test:
-    #  - onCartClear from cart
-    #  - onProductAdd from catalog, product and order pages
-    #  - onProductRemove from cart and order page
+    #  - onProductAdd from catalog and order pages
+    #  - onProductRemove from order page
 
-    @property
-    def reached_goals(self):
+    def tearDown(self):
+        # delete the session to clear the cart
+        self.browser.delete_cookie('sessionid')
+        self.browser.execute_script('localStorage.clear();')
+        super().setUp()
+
+    def get_goals(self):
         return self.browser.execute_script('return window.dataLayer.results;')
 
-    def get_first(self, reached):
+    def get_goal(self, reached, index=0):
         """Get the first reached goal and unfold it."""
```

Comment:
1. """Get a reached with the given index and unfold it."""

2. And it'll be good to add typing to `get_goal` method

3. We should create separated class Goals (or [FakeGoals](https://www.yegor256.com/2015/05/25/unit-test-scaffolding.html)). `goals.get(index: int)` would be more clear way. Pdd subtask would be good for it

Reply:
>And it'll be good to add typing to get_goal method

It has complex and heterogeneous structure, so I couldn't call it a hint. We already have a link to docs with structure specification

>We should create separated class Goals (or FakeGoals). goals.get(index: int) would be more clear way. Pdd subtask would be good for it

I believe it can't be "fake" because we have no any Goal's interface to fake.
But I agree it will more convenient to operate with Goals class